pkg/backend/display/object_diff.go

```diff
@@ -692,10 +701,39 @@ func shortHash(hash string) string {
 func printOldNewDiffs(
 	b *bytes.Buffer, olds resource.PropertyMap, news resource.PropertyMap, include []resource.PropertyKey,
 	planning bool, indent int, op display.StepOp, summary bool, truncateOutput bool, debug bool, showSecrets bool,
+	hidePaths []resource.PropertyPath,
 ) {
+	var hiddenDiffs []resource.PropertyPath
+
 	// Get the full diff structure between the two, and print it (recursively).
-	if diff := olds.Diff(news, resource.IsInternalPropertyKey); diff != nil {
-		PrintObjectDiff(b, *diff, include, planning, indent, summary, truncateOutput, debug, showSecrets)
+	diff := olds.DiffWithOptions(news,
+		resource.IgnoreKeyFunc(resource.IsInternalPropertyKey),
+		resource.IgnorePathFunc(func(path resource.PropertyPath) bool {
+			for _, v := range hidePaths {
+				if v.Contains(path) {
+					hiddenDiffs = append(hiddenDiffs, v)
+					return true
+				}
+			}
+			return false
+		}),
+	)
+
+	// Ensure that our paths are unique and sorted
+	slices.SortFunc(hiddenDiffs, func(a, b resource.PropertyPath) int {
+		return cmp.Compare(a.String(), b.String())
+	})
+	hiddenDiffs = slices.CompactFunc(hiddenDiffs, func(a, b resource.PropertyPath) bool {
+		return a.String() == b.String()
+	})
+
+	// We have hidden all the diffs, but there was a diff.
+	if len(hiddenDiffs) > 0 {
```

Comment:
Is this right?  Doesn't this overwrite `diff` if there are *any* hidden diffs? I think this should be `len(hiddenDiffs) > 0 && diff == nil`